ui/pages/reservation-unit/[id].tsx

```diff
@@ -361,14 +363,23 @@ const ReservationUnit = ({
   const isMobile = useMedia(`(max-width: ${breakpoints.m})`, false);
 
   const router = useRouter();
+  const session = useSession();
+  const dynamicRoute = router.asPath;
+
   const [, setPendingReservation] = useSessionStorage(
     "pendingReservation",
     null
   );
 
   const now = useMemo(() => new Date().toISOString(), []);
 
-  const { reservation, setReservation } = useContext(DataContext);
+  const { reservation: contextReservation, setReservation } =
+    useContext(DataContext);
+
+  const reservation =
+    contextReservation?.reservationUnitPk === reservationUnit.pk
+      ? contextReservation
+      : null;
```

Comment:
I guess this is fine, but this should be more an assert (should never happen) and the context should be cleared when user switches between ReservationUnits. Alternatively the partial reservation (context) should be shown to the user and asked if he wants to replace it with the new one.
Or when user navigates outside the ReservationUnit confirm and clear context.

Reply:
Yes, this inspired me to clean up some sedimented logic.